src/black/__init__.py

```diff
@@ -1495,8 +1369,14 @@ def patch_click() -> None:
 
 
 def patched_main() -> None:
-    maybe_install_uvloop()
-    freeze_support()
+    global __BLACK_MAIN_CALLED__
+    __BLACK_MAIN_CALLED__ = True
```

Comment:
Instead of this global, what if we return early in `maybe_install_uvloop` if uvloop is already in sys.modules?

Or we could just call it unconditionally. It's only called once per formatting run and it's pretty fast:

```
In [14]: %timeit black.concurrency.maybe_install_uvloop()
1.67 µs ± 30.3 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```